flink-scala/src/test/scala/org/apache/flink/api/operator/SelectByFunctionTest.scala

```diff
@@ -0,0 +1,233 @@
+      Assert.assertSame("SelectByMax must return smaller tuple",
+        smaller, maxByTuple.reduce(smaller, smaller))
+    } catch {
+      case e : Exception => Assert.fail("No exception should be thrown" +
+        " while comapring both tuples")
+    }
+  }
+
+  // ----------------------- MINIMUM FUNCTION TEST BELOW --------------------------
+
+  /**
+    * This test validates whether the order of tuples has any impact
+    * on the outcome and if the smaller tuple is returned.
+    */
+  @Test
+  def testMinByComparison() : Unit = {
+    val a1 = Array(0)
+    val minByTuple = new SelectByMinFunction(tupleTypeInfo, a1)
+    try {
+      Assert.assertSame("SelectByMin must return smaller tuple",
+        smaller, minByTuple.reduce(smaller, bigger))
+      Assert.assertSame("SelectByMin must return smaller tuple",
+        smaller, minByTuple.reduce(bigger, smaller))
+    } catch {
+      case e : Exception => Assert.fail("No exception should be thrown " +
+        "while comapring both tuples")
+    }
+  }
+
+  /**
+    * This test cases checks when two tuples only differ in one value, but this value is not
+    * in the fields list. In that case it should be seen as equal and
+    * then the first given tuple (value1) should be returned by reduce().
+    */
+  @Test
+  def testMinByComparisonSpecialCase1() : Unit = {
+    val a1 = Array(0, 3)
+    val minByTuple = new SelectByMinFunction(tupleTypeInfo, a1)
+
+    try {
+      Assert.assertSame("SelectByMin must return the first given tuple",
+        specialCaseBigger, minByTuple.reduce(specialCaseBigger, bigger))
+      Assert.assertSame("SelectByMin must return the first given tuple",
+        bigger, minByTuple.reduce(bigger, specialCaseBigger))
+    } catch {
+      case e : Exception => Assert.fail("No exception should be thrown " +
+        "while comapring both tuples")
+    }
+  }
+
+  /**
+    * This test validates that when two tuples only differ in one value
+    * and that value's index is given at construction time. The smaller tuple must be returned
+    * then.
+    */
+  @Test
+  def  testMinByComparisonSpecialCase2() : Unit = {
+    val a1 = Array(0, 2, 1, 4, 3)
+    val minByTuple = new SelectByMinFunction(tupleTypeInfo, a1)
+
+    try {
+      Assert.assertSame("SelectByMin must return smaller tuple",
+        smaller, minByTuple.reduce(specialCaseSmaller, smaller))
+      Assert.assertSame("SelectByMin must return smaller tuple",
+        smaller, minByTuple.reduce(smaller, specialCaseSmaller))
+    } catch {
+      case e : Exception => Assert.fail("No exception should be thrown" +
+        " while comapring both tuples")
+    }
+  }
+
+  /**
+    * Checks whether reduce does behave as expected if both values are the same object.
+    */
+  @Test
+  def testMinByComparisonMultiple() : Unit =  {
+    val a1 = Array(0, 1, 2, 3, 4)
+        val minByTuple : SelectByMinFunction[scala.Tuple5[Int, Long, String, Long, Int]] =
```

Comment:
indention + change to `val minByTuple = new SelectByMinFunction(tupleTypeInfo, a1)`